### DISCUSSION TITLE
Passing 0 as for NULL System.Object says parameter types mismatch

### DISCUSSION THREAD

**Discussion by aimardcr** [vfsfitvnm/frida-il2cpp-bridge#253] [Q&A]
So this function has default parameter, but i still have to pass values to the parameter to able to invoke it. But when i pass `0` as the parameter, it always says `il2cpp: cannot invoke method Active: parameter types mismatch`.
I can confirm that the first param works fine because i tried passing the FrameID function to other method and it works fine.

Function:
```
BaseFrame Active(FrameID frameID, System.Object arg, System.Boolean bEff, eUIPrioType prioType); // 0xffffffffc4c756b4
```

Code:
```
        const mainAssembly = Il2Cpp.Domain.assembly('Assembly-CSharp').image;
        const mscorlib = Il2Cpp.Image.corlib;

        const Mgr = mainAssembly.class('Mgr');
        const m_Instance = Mgr.field('m_Instatnce').value as Il2Cpp.Object;
        const m_UI = m_Instance.field('m_UI').value as Il2Cpp.Object;

        const FrameID = mainAssembly.class('FrameID');
        m_UI.method('Active').invoke(FrameID.field('FRAME_CustomRoomMode').value, 0, false, 1);
```

I tried create new System.Object, but it still also says invalid parameter. The last param is enum but i assume passing normal integer is fine right?

**Comment by vfsfitvnm** [comment]
> I tried create new System.Object, but it still also says invalid parameter

You need to pass a [`NativePointerValue`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/frida-gum/index.d.ts#L1624), so either Frida's `NULL` or `new Il2Cpp.Object(NULL)` are fine.

> The last param is enum but i assume passing normal integer is fine right?

Nope! See https://github.com/vfsfitvnm/frida-il2cpp-bridge/discussions/125#discussioncomment-2219280

**Reply by aimardcr** [reply]
Ah yes, i was wrong about the enum. Thank you, solved!
